.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "test/testdata/golden/googleapis"]
```

Comment:
This is a little bit weird. In fact, it is weird enough that I am pretty sure it is not what you want. You have _both_ a submodule and an explicit `git clone`. And the way you have them written are not equivalent.

`git submodule` tags a repository as a submodule to the outer repository, and the sub-repository is grafted in _at a specific commit_. That is why if you look farther down in this "files changed" view, you will see this:

![screenshot from 2017-07-28 07 59 30](https://user-images.githubusercontent.com/4346/28723163-bd1f5746-736a-11e7-8f1b-be228631817b.png)

Notice how it says "added at 75c3a5". The repository is grafted in _at a specific point_.
Your `git clone` command, however, (1) plows over the graft, and (2) will clone HEAD. This is going to give you a repository in an inconsistent state. In some ways this is fine; CircleCI is never going to commit the result so it will hit the bit bucket as soon as the VM spins down. On the other hand, this inconsistency is likely to cause a decent bit of heartburn when contrasting local testing with CI as the gulf between 75c3a5 and HEAD widen.

I can not tell from reading this PR _which_ behavior you actually want. If your goal is to always use googleapis pinned to a specific point, then a submodule _might_ be a good answer. If it is (holding off on discussing that until I understand your goals), at minimum your `git clone` above needs to be replaced with `git submodule update --force`.

If your goal is to use googleapis at HEAD, submodules are very much the wrong tool. You would be far better off just `git clone`-ing the repo to a designated location. As I will say below, this will cause problems for local testing, though, which need to be talked through.